modules/sdk-coin-eth/src/lib/utils.ts

```diff
@@ -552,15 +562,29 @@ export function hasSignature(txData: TxData): boolean {
   );
 }
 
+type RecursiveBufferOrString = string | Buffer | RecursiveBufferOrString[];
+
 /**
  * Get the raw data decoded for some types
  *
  * @param {string[]} types ABI types definition
  * @param {Buffer} serializedArgs encoded args
  * @returns {Buffer[]} the decoded raw
  */
-export function getRawDecoded(types: string[], serializedArgs: Buffer): Buffer[] {
-  return EthereumAbi.rawDecode(types, serializedArgs);
+export function getRawDecoded(types: string[], serializedArgs: Buffer): RecursiveBufferOrString[] {
```

Comment:
The library underneath did not change, but the signature here was all wrong, and the old ethUtils were very lax about types and did a lot of auto-conversions that made up for it.

Reply:
I don't know. Then we are going to be converting everything to string, just to unconvert it to Buffer again in many cases etc.

At the core it's you decode bunch of data from a binary of transactions. You pass parameter that specify the types of data you expect. And then you get back your data in corresponding types that are guaranteed to be of the right type.

This normalize function stuff there is to fix incorrect type signature that was lying before (and we were getting away with it because this old eth lib was doing a lot of auto-conversion everywhere, and then figured out that it's a stupid idea (which is true)), and we we couldn't upgrade to new library because of that.

As long as we keep this weird and old `ethereum-abi` library, I think it's OK to just specify that `string | buffer` is returned and let the caller pick, the type that it knows the return value is in.